{
  FindX findX=new FindX();
  assertEquals((char)retryGettingResultOrFailing(findX,"hexy",1000,"shouldn't happen"),'x');
  boolean secondRetrySucceeds=false;
  try {
    retryGettingResultOrFailing(findX,"hexy",100,"expected");
    secondRetrySucceeds=true;
  }
 catch (  AssertionError e) {
    assertTrue(e.toString().contains("expected"));
  }
  if (secondRetrySucceeds)   fail("should have thrown");
  assertNotNull(findX.getLastFailure());
  assertFalse(findX.getLastFailure().toString().contains("expected"));
}
